components/raftstore/src/store/worker/pd.rs

```diff
@@ -1969,6 +1990,31 @@ where
         (interval_second >= self.store_heartbeat_interval.as_secs())
             && (interval_second <= STORE_HEARTBEAT_DELAY_LIMIT)
     }
+
+    fn handle_control_grpc_server(&mut self, event: pdpb::ControlGrpcEvent) {
+        info!("forcely control grpc server";
+                "curr_health_status" => ?self.curr_health_status,
+                "event" => ?event,
+        );
+        match event {
+            pdpb::ControlGrpcEvent::Pause => {
+                if let Err(e) = self.grpc_service_manager.pause() {
+                    warn!("failed to send service event to PAUSE grpc server";
+                            "err" => ?e);
+                } else {
+                    self.update_health_status(ServingStatus::NotServing);
+                }
+            }
+            pdpb::ControlGrpcEvent::Resume => {
+                if let Err(e) = self.grpc_service_manager.resume() {
+                    warn!("failed to send service event to RESUME grpc server";
+                            "err" => ?e);
+                } else {
+                    self.update_health_status(ServingStatus::Serving);
+                }
+            }
```

Comment:
How much time do they take? Could you add to log?

Reply:
Maybe it's redundant? `pause` and `resume` just sends messages to the external receiver, and it's a light-weight operation.

Reply:
Oh, I see, is there any log after the actual pause and resume operation?

Reply:
> Oh, I see, is there any log after the actual pause and resume operation?

Yep, as @tonyxuqqi suggested, the loggings of PAUSE | RESUME operation (containing the time costs ) have been supplied in the latest commit.